Sources/MapboxMobileEvents/NSUserDefaults+MMEConfiguration.m

```diff
@@ -317,12 +318,38 @@ - (NSString *)mme_userAgentString {
              [NSProcessInfo mme_processorTypeDescription]
         ];
         
+        // try to get a list of loaded dylibs
+        struct task_dyld_info tdi;
+        mach_msg_type_number_t count = TASK_DYLD_INFO_COUNT;
+        if (KERN_SUCCESS != task_info(mach_task_self_, TASK_DYLD_INFO, (task_info_t)&tdi, &count)) {
+            return;
+        }
+
+        struct dyld_all_image_infos *infos = (struct dyld_all_image_infos *)tdi.all_image_info_addr;
+        struct dyld_image_info *infoArray = (struct dyld_image_info *)infos->infoArray;
+
+        NSString* mainBundleIdentifier = NSBundle.mme_mainBundle.bundleIdentifier;
+
         // check all loaded frameworks for mapbox frameworks, record their bundleIdentifier
         NSMutableSet *loadedMapboxBundleIds = NSMutableSet.new;
-        for (NSBundle *loaded in [NSBundle.allFrameworks arrayByAddingObjectsFromArray:NSBundle.allBundles]) {
-            if ([loaded.bundleIdentifier hasPrefix:@"com.mapbox"]
-                && loaded.bundleIdentifier != NSBundle.mme_mainBundle.bundleIdentifier) {
-                [loadedMapboxBundleIds addObject:loaded.bundleIdentifier];
+        for (uint32_t i = 0; i < infos->infoArrayCount; ++i) {
+            struct dyld_image_info *imageInfo = infoArray + i;
+
+            NSString* path = [NSString stringWithCString:imageInfo->imageFilePath encoding:NSUTF8StringEncoding];
+            // check if a framework located inside app's bundle, only those can be ours
+#if !TARGET_OS_SIMULATOR
+            NSString* mainBundlePath = NSBundle.mainBundle.bundlePath;
+            if ([path hasPrefix:mainBundlePath]) {
```

Comment:
Why doesn't this work on simulator?

Reply:
From our experience, on simulator Frameworks are loaded not from `<app_bundle>/Frameworks`, but directly from build directories for some reason. So filtering by app_bundle path is not possible. Not sure if there is some issue with project settings or does it work as intended.

Reply:
We are more interested on optimizing the method on real devices, so decided to go `#ifdef` way here.